### PR TITLE
basic models

### PR DESCRIPTION
**Pull request by kolyastepanets**

### CODE REVIEW

File: spec/factories/shareconomy_listings.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+  factory :shareconomy_listing, class: 'Shareconomy::Listing' do
+    contact_name "MyString"
+    service_name "MyString"
+    phone_number "MyString"
+    max_duration "2016-06-02 17:17:48"
+    max_guests 1
+    instant_booking false
+  end
```
Comment (kopchak) [inline spec/factories/shareconomy_listings.rb]: faker?

File: spec/factories/shareconomy_packages.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+  factory :shareconomy_package, class: 'Shareconomy::Package' do
+    name "MyString"
+    description "MyText"
+    time_duration "2016-06-02 17:18:54"
+    price "9.99"
+    deposit_amount "9.99"
+    listing_id 1
```
Comment (kopchak) [inline spec/factories/shareconomy_packages.rb]: same

**Comment by kopchak** [comment]
ok

File: app/models/shareconomy/listing.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module Shareconomy
+  class Listing < ActiveRecord::Base
+    belongs_to :user
+    has_many :packages
+
+    validates :instant_booking, inclusion: { in: [true, false] }
```
Comment (GeorgeZhukov) [inline app/models/shareconomy/listing.rb]: what about validates user presence?

File: app/models/shareconomy/order.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+module Shareconomy
+  class Order < ActiveRecord::Base
+    include AASM
+
+    belongs_to :package
+    belongs_to :buyer, class_name: User.name
+    belongs_to :seller, class_name: User.name
```
Comment (GeorgeZhukov) [inline app/models/shareconomy/order.rb]: no validations?

File: app/models/shareconomy/package.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module Shareconomy
+  class Package < ActiveRecord::Base
+    belongs_to :listing
+    has_many :orders
+
```
Comment (GeorgeZhukov) [inline app/models/shareconomy/package.rb]: can be package without listing?

Comment (sparrow) [inline app/models/shareconomy/package.rb]: Why do we need package? @GeorgeZhukov, @kolyastepanets, could you please provide a final model's structure first?

File: config/routes.rb [hunk 6]
```diff
@@ -1,2 +1,3 @@
 Shareconomy::Engine.routes.draw do
+ mount_devise_token_auth_for Shareconomy::User.name, at: 'api/v1/auth'
```
Comment (GeorgeZhukov) [inline config/routes.rb]: check spaces pls

File: spec/factories/shareconomy_messages.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :shareconomy_message, class: 'Shareconomy::Message' do
+    sender_id 1
+    recipient_id 1
+    body "MyText"
```
Comment (GeorgeZhukov) [inline spec/factories/shareconomy_messages.rb]: Faker?

File: spec/factories/shareconomy_orders.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+FactoryGirl.define do
+  factory :shareconomy_order, class: Shareconomy::Order.name do
```
Comment (GeorgeZhukov) [inline spec/factories/shareconomy_orders.rb]: why empty?

File: app/models/shareconomy/message.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+module Shareconomy
+  class Message < ActiveRecord::Base
+    belongs_to :recipient, class_name: User.name, inverse_of: :received_messages
```
Comment (GeorgeZhukov) [inline app/models/shareconomy/message.rb]: what about validation?

**Comment by GeorgeZhukov** [comment]
ok

**Pull request merged**